app/models/user_status.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class UserStatus < ActiveRecord::Base
+  belongs_to :user
+
+  validate :ends_at_greater_than_set_at
```

Comment:
I think we will only need to run this validation when `ends_at` and `set_at` are going to change.

Reply:
I'm not sure here. Do you mean that it's better to remove the validator for now and then return it back to this class when we'll add code that will be setting `ends_at`? Or do you mean that this shouldn't be implemented as a validator on the model, and we need to check it at the place where we set `ends_at`?

Reply:
O sorry I mean that this validation will only need to run when `will_save_change_to_ends_at` or `will_save_change_to_set_at` is `true`.

Reply:
Ok, got it. I've added a condition to the validator.